activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -596,8 +605,9 @@ def reconnect!(restore_transactions: false)
           end
         rescue => original_exception
           translated_exception = translate_exception_class(original_exception, nil, nil)
+          retry_deadline_exceeded = deadline && deadline < Process.clock_gettime(Process::CLOCK_MONOTONIC)
 
-          if retries_available > 0
+          if !retry_deadline_exceeded && retries_available > 0
```

Comment:
Doesn't matter here, but since you are likely to refactor this in your followup, I figured I'd mention it.

As a general rule, when testing multiple conditions, it's preferable to perform the faster checks first and the more expensive ones later.

In this case, `retries_available > 0` is a very simple integer comparision, while checking the deadline requires to do a syscall through `Process.clock_gettime(Process::CLOCK_MONOTONIC)`.

Again, not that it matters much in the grand scheme of things, especially here, but it's just a good general practice.

Reply:
Makes a lot of sense, thanks Jean! ❤️ I can flip the ordering when I do the refactor so we avoid the syscall if we're out of retries anyway.